### PR TITLE
feat(room-metadata) add module for generic room metadata storage

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: resources/prosody-plugins/mod_muc_room_metadata.lua [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+local room_metadata_muc_component_config = module:get_option_string('room_metadata_muc', 'metadata.'..module.host);
+
+module:depends('jitsi_session');
+
+local component_muc_service;
+local main_muc_service;
+
+
+-- Utility functions
+
```
Comment (damencho) [inline resources/prosody-plugins/mod_muc_room_metadata.lua:44]: Yeah maybe leave a comment that is an optimization step so clients don't need to be notified via separate message about it on join.

**Comment by damencho** [comment]
LGTM

File: resources/prosody-plugins/mod_muc_room_metadata.lua (hunk 1 above)
Comment (saghul) [reply resources/prosody-plugins/mod_muc_room_metadata.lua:44]: Done.

File: resources/prosody-plugins/mod_muc_room_metadata.lua [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+-- It needs a room metadata muc component
+--
+-- VirtualHost "jitmeet.example.com"
+--     modules_enabled = {
+--         "muc_room_metadata"
+--     }
+--     room_metadata_muc = "metadata.jitmeet.example.com"
+--     main_muc = "muc.jitmeet.example.com"
+--
+-- Component "metadata.jitmeet.example.com" "muc"
```
Comment (damencho) [inline resources/prosody-plugins/mod_muc_room_metadata.lua:13]: Why do we need this to be a muc? That just gets messages and should be like speaker stats component, does not need to be a muc component, right?

**Comment by damencho** [comment]
What about enabling it by default in the debian config?

**Review by damencho — APPROVED** [review]

**Pull request merged**